### PR TITLE
Implement for loop with slice/subscript objects

### LINKED ISSUE #883: Implement looping through a reversed string
```console
(lp) namannimmo@srf lpython % cat t.py
def f():
    s: str
    s = "abcde"
    c: str
    res: str
    for c in s[::-1]:
        print(c)


f()
(lp) namannimmo@srf lpython % python t.py
e
d
c
b
a
(lp) namannimmo@srf lpython % lpython t.py && ./a.out 
semantic error: Only function call `range(..)` supported as for loop iteration for now
 --> t.py:6:5 - 7:16
  |
6 |        for c in s[::-1]:
  |        ^^^^^^^^^^^^^^^^^...
...
  |
7 |            print(c)
  | ...^^^^^^^^^^^^^^^^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
(lp) namannimmo@srf lpython %
```

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #883

### CODE REVIEW

File: integration_tests/test_builtin_str.py [hunk 1]
```diff
@@ -63,6 +63,15 @@ def test_str_slice_step():
     assert s[::-1] == "kjihgfedcba"
 
 
+def test_issue_883():
+    s: str
+    s = "abcde"
+    c: str
+    for c in s[::-1]:
+        print(c)
```
Comment (czgdp1807) [inline integration_tests/test_builtin_str.py:71]: This looks okay. Do we support `reversed` intrinsic as well?

Comment (czgdp1807) [inline integration_tests/test_builtin_str.py:71]: This looks okay. Do we support `reversed` intrinsic as well?

Comment (Smit-create) [reply integration_tests/test_builtin_str.py:71]: Not yet.

Comment (czgdp1807) [reply integration_tests/test_builtin_str.py:71]: Okay. Makes sense. Let's support that in a subsequent PR.

**Review by czgdp1807 — APPROVED** [review]
@certik Is this good to merge?

**Review by certik — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**